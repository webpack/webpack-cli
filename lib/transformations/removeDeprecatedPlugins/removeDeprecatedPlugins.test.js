const defineTest = require('../defineTest');

defineTest(__dirname, 'removeDeprecatedPlugins', 'removeDeprecatedPlugins-0');
defineTest(__dirname, 'removeDeprecatedPlugins', 'removeDeprecatedPlugins-1');
defineTest(__dirname, 'removeDeprecatedPlugins', 'removeDeprecatedPlugins-2');
defineTest(__dirname, 'removeDeprecatedPlugins', 'removeDeprecatedPlugins-3');
defineTest(__dirname, 'removeDeprecatedPlugins', 'removeDeprecatedPlugins-4');
