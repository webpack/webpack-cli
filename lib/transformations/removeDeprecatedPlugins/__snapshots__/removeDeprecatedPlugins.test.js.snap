// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeDeprecatedPlugins transforms correctly using "removeDeprecatedPlugins-0" data 1`] = `
"// Works for OccurrenceOrderPlugin
module.exports = {}
"
`;

exports[`removeDeprecatedPlugins transforms correctly using "removeDeprecatedPlugins-1" data 1`] = `
"// Works for DedupePlugin
module.exports = {}
"
`;

exports[`removeDeprecatedPlugins transforms correctly using "removeDeprecatedPlugins-2" data 1`] = `
"// Doesn't remove unmatched plugins
module.exports = {
    plugins: [new webpack.optimize.UglifyJsPlugin()]
}
"
`;

exports[`removeDeprecatedPlugins transforms correctly using "removeDeprecatedPlugins-3" data 1`] = `
"// This should throw
export default (config) => {
    config.plugins.push(new webpack.optimize.UglifyJsPlugin());
    config.plugins.push(new webpack.optimize.DedupePlugin());
    config.plugins.push(new webpack.optimize.OccurrenceOrderPlugin());
    return config
}
"
`;

exports[`removeDeprecatedPlugins transforms correctly using "removeDeprecatedPlugins-4" data 1`] = `
"// This should throw
const inst = new webpack.optimize.OccurrenceOrderPlugin()
export default (config) => {
    config.plugins = [
        inst
    ]
    return config
}
"
`;
